src/main/java/gregtech/api/capability/IPropertyFluidFilter.java

```diff
@@ -0,0 +1,80 @@
+package gregtech.api.capability;
+
+import gregtech.api.fluids.MaterialFluid;
+import gregtech.api.fluids.fluidType.FluidType;
+import gregtech.api.fluids.fluidType.FluidTypes;
+import net.minecraftforge.fluids.Fluid;
+import net.minecraftforge.fluids.FluidStack;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Fluid filter based on fluid properties; i.e. temperature, fluid state, and various material flags such as acid
+ * and plasma.
+ *
+ * @see FluidType
+ * @see FluidTypes
+ * @see MaterialFluid
+ */
+public interface IPropertyFluidFilter extends IFilter<FluidStack> {
+
+    /**
+     * Minimum temperature of the fluid in kelvin before it starts being considered 'cryogenic'; if a fluid has lower
+     * temperature than this, it's considered cryogenic.
+     */
+    int CRYOGENIC_TEMPERATURE_THRESHOLD = 120;
+
+    @Override
+    @SuppressWarnings("RedundantIfStatement")
+    default boolean test(@Nonnull FluidStack stack) {
+        Fluid fluid = stack.getFluid();
+        int temperature = fluid.getTemperature();
+        if (temperature > getMaxFluidTemperature()) return false;
```

Comment:
The exception for plasma containers seems to be lost here.

Plasma containers can hold any plasma, even if the temperature is higher than the pipe maximum, but will still fail to contain other non-plasma fluids which are too hot.

Reply:
That's weird, because this logic was copied from `IThermalFluidHandlerItemStack` with pretty much zero modification. Is the exception being omitted there as well?